pravega-k8-log-collection/pravega-k8s-fluentbit-log.sh

```diff
@@ -0,0 +1,95 @@
+#/usr/bin/env bash
+#
+# Pravega K8 based log collection tool
+#
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+set -euo pipefail
+# Setting up temporary log collection directory under /tmp/logdir
+mkdir -p ./logdir
+output_dir="./logdir"
+# Capturing pod, services and persistentvolumeclaim information from all namespaces
+kubectl get po,svc,pvc --all-namespaces > $output_dir/kubectl-get-po-svc-pvc-all-namespace_output
+IFS=$'\n'
+# We now rok only on the "fluentbit" pods that contain the logs of the rest of services.
+for i in $(cat $output_dir/kubectl-get-po-svc-pvc-all-namespace_output | grep -i fluent-bit)
+do
+
+# Detect Pravega pods information in various K8 deployment (e.g. PKS / GKE)
+if [[ $i == *"po/"* ]] || [[ $i == *"pod/"* ]]; then
+
+    # Collect logs from "Running" pods, if pods are not in "Running" state it will collect describe information.
+    if [ "`echo $i | awk '{print $4}'`" == "Running" ]; then
+    {
+        fluentbit_pod=$(echo $i |  awk '-F'[/] '{print $2}'| awk '{print $1}')
+        namespace=$(echo $i | awk '{print $1}')
+        # Clean any pre-existing tar file in the fluentbit pod.
+        kubectl -n $namespace exec $fluentbit_pod -- bash -c "rm /tmp/$fluentbit_pod.tar.gz" || true
+        # Create a tar.gz file with all the logs contained in the fluentbit pod.
+        set +e
+        kubectl -n $namespace exec $fluentbit_pod -- bash -c "tar -czf /tmp/$fluentbit_pod.tar.gz /var/vcap/store/docker/docker/containers"
+        exitcode=$?
+        # Some logs may be being written during collection, which makes tar to exit with a non-zero code. Need to handle this case.
```

Comment:
We could use `--ignore-failed-read ` option of tar to solve the error that logs may be being written during collection.

Reply:
Tried with that flag, but `tar` seems to continue exiting with with non-zero code. That is, using:
`kubectl -n $namespace exec $fluentbit_pod -- bash -c "tar --ignore-failed-read -czf /tmp/$fluentbit_pod.tar.gz /var/vcap/store/docker/docker/containers"`
If we remove the non-zero exit code handling, we get the following error:
```
tar: /var/vcap/store/docker/docker/containers/43b58b96ef7f55459261247c397acfa72e26b3e7c055909ea199ed7df4aa3d43/43b58b96ef7f55459261247c397acfa72e26b3e7c055909ea199ed7df4aa3d43-json.log: file changed as we read it
command terminated with exit code 1
```
I assume that the `--ignore-failed-read` is supposed to help us to remove the non-zero exit code handling, or maybe I'm totally wrong here.